File: algorithms/strings/alternating-characters/alternating-characters-java/alternating-characters-java/src/TesteAlternatingCharacters.java
import static org.junit.Assert.assertEquals;

import org.junit.Test;

public class TesteAlternatingCharacters {

    @Test
    public void testeCaso1() {

        String s = "AAAA";
        int esperado = 3;

        int resultado = Resultado.alternatingCharacters(s);

        assertEquals(esperado, resultado);
    }
    
    @Test
    public void testeCaso2() {

        String s = "ABABABAB";
        int esperado = 0;

        int resultado = Resultado.alternatingCharacters(s);

        assertEquals(esperado, resultado);
    }
}

File: algorithms/strings/alternating-characters/alternating-characters-javascript/js/testeAlternatingCharacters.js
import {alternatingCharacters} from '../js/alternatingCharacters.js';
import assert from 'assert';

describe('alternatingCharacters', function() {

    it('O número mínimo de exclusões necessárias é: 3', function() {

        let s = 'AAAA';
        let esperado = 3;

        let resultado = alternatingCharacters(s);

        assert.strictEqual(resultado, esperado);

    });
});

describe('alternatingCharacters', function() {

    it('O número mínimo de exclusões necessárias é: 0', function() {

        let s = 'ABABABAB';
        let esperado = 0;

        let resultado = alternatingCharacters(s);

        assert.strictEqual(resultado, esperado);

    });
});

describe('alternatingCharacters', function() {

    it('O número mínimo de exclusões necessárias é: 4', function() {

        let s = 'AAABBB';
        let esperado = 4;

        let resultado = alternatingCharacters(s);

        assert.strictEqual(resultado, esperado);

    });
});